modules/primo/src/Primo/Ting/Object.php

```diff
@@ -253,22 +344,29 @@ public function getVersion() {
   /**
    * @inheritDoc
    */
-  public function isPartOf() {
-    // TODO: Implement isPartOf() method.
+  public function getLanguage() {
+    $lang = $this->document->getLanguage();
+    if (!empty($lang)) {
+      // Languages returned by Primo is in ISO-639 format. To return something
+      // understandable by users we return the language for the code in the
+      // native tounge.
+      $languageConverter = new ISO639();
+      return $languageConverter->nativeByCode3($lang);
```

Comment:
This wil print the language in the native code so "Íslenska" for icelandic, "italiano" for italian etc. I wonder whether it would be more appropriate to use the English version of the language and then pass it through `t()` to translate to the active Drupal language.

Reply:
Not a bad idea!